packages/angular_devkit/build_angular/src/angular-cli-files/plugins/single-test-transform.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { logging } from '@angular-devkit/core';
+import { getOptions } from 'loader-utils';
+import { loader } from 'webpack';
+
+export interface SingleTestTransformLoaderOptions {
+  files: string[];
+  logger: logging.Logger;
+}
+
+export default function loader(this: loader.LoaderContext, source: string) {
+  const options = getOptions(this) as SingleTestTransformLoaderOptions;
+  const start = 'import \'';
+  const end = '\';';
+  const testCode = start + options.files
+    .map(path => `./${path.replace('.ts', '')}`)
+    .join(`${end}\n${start}`) + end;
+
+  // TODO: maybe a documented 'marker/comment' inside test.ts would be nicer?
+  let mockedRequireContext = '{ keys: () => ({ map: (_a) => { } }) };';
+  mockedRequireContext += process.platform === 'win32' ? '\r\n' : '\n';
+  source = source.replace(/require\.context\(.*/, mockedRequireContext + testCode);
```

Comment:
If we can't replace the file, we should bail out and offer an informative message instead. This way the user isn't left wondering why the command isn't working.

A good error message would be something along these lines:
```
Tried to transform main file for tests but could not find expected code.

The 'spec' option requires that the 'main' file for tests include the line below:
  const context = require.context('./', true, /\.spec\.ts$/);

Please ensure this line is present to use the `spec` option.  
```